client/services/utils/localSettings.js

```diff
@@ -0,0 +1,143 @@
+    const listeners = this.listeners
+    if (this.settingsObject.hasOwnProperty(sourceName)) {
+      const existingSettings = this.settingsObject[sourceName]
+      settingsArray.forEach(setting => {
+        const settingForUpdate = existingSettings.find(entry => {
+          return entry.name === setting.name
+        })
+        if (settingForUpdate) {
+          const idx = settingsObject[sourceName].indexOf(settingForUpdate)
+          settingsObject[sourceName][idx][VALUE_KEY] = setting[VALUE_KEY]
+        } else {
+          settingsObject[sourceName].push(setting)
+        }
+      })
+    } else {
+      settingsObject[sourceName] = settingsArray
+    }
+    settingsArray.forEach(setting => {
+      if (!listeners.hasOwnProperty(`${sourceName}-${setting.name}`)) {
+        listeners[`${sourceName}-${setting.name}`] = setting.cb
+      }
+    })
+    this.listeners = listeners
+    this.settingsObject = settingsObject
+    this.$window.sessionStorage.setItem(this.STORAGE_KEY, JSON.stringify(this.settingsObject))
+  }
+
+  update (settings, sourceName, changeCallback) {
+    const listeners = this.listeners
+    const settingsObject = this.settingsObject
+    this.checkSettingValidity(settings)
+    Object.keys(settings).forEach(key => {
+      if (settingsObject.hasOwnProperty(key) && Array.isArray(settings[key])) {
+        if (Array.isArray(settings[key])) {
+          settings[key].forEach(item => {
+            const forcb = settingsObject[key].find(existing => {
+              return existing.name === item.name
+            })
+            if (forcb) {
+              const idx = settingsObject[key].indexOf(forcb)
+              settingsObject[key][idx][VALUE_KEY] = item[VALUE_KEY]
+              forcb[VALUE_KEY] = item[VALUE_KEY]
+            }
+          })
+        } else if (typeof (settings[key]) === 'object') {
+          const constructArray = []
+          Object.keys(settings[key]).forEach(givenSetting => {
+            constructArray.push({
+              name: `${givenSetting}`,
+              dataType: typeof (settings[key][givenSetting]),
+              value: settings[key][givenSetting],
+              cb: changeCallback
+            })
+            listeners[`${sourceName}-${givenSetting}`] = changeCallback
+          })
+          settingsObject[key].concat(constructArray)
+        }
+      } else {
+        this.addSettings(key, settings[key])
+      }
+    })
+    Object.assign(this.settingsObject, settingsObject)
+    this.$window.sessionStorage.setItem(this.STORAGE_KEY, JSON.stringify(this.settingsObject))
+    Object.keys(settings).forEach(key => {
+      if (Array.isArray(settings[key])) {
+        settings[key].forEach(item => {
+          if (listeners.hasOwnProperty(`${key}-${item.name}`)) {
+            listeners[`${key}-${item.name}`]()
+          }
+        })
+      }
+    })
+  }
+
+  get (sourceName) {
+    const sessionSettings = JSON.parse(this.$window.sessionStorage.getItem(this.STORAGE_KEY))
```

Comment:
if you are talking about a situation where `this.$window.sessionStorage` is undefined, then yes, it will error.

but if you are talking about a situation where the storage just doesn't have the `localsettings` key, then it won't error.

the `getItem` will return `null`, and `JSON.parse(null)` gives `null`, which is checked for in the next line.